c7n/resources/vpc.py

```diff
@@ -1600,3 +1600,73 @@ class resource_type(object):
         name = 'KeyName'
         date = None
         dimension = None
+
+
+@Vpc.action_registry.register('set-vpc-flow')
+@Subnet.action_registry.register('set-vpc-flow')
+@NetworkInterface.action_registry.register('set-vpc-flow')
+class CreateFlowLogs(BaseAction):
+    """Create flow logs for a network resource
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: vpc-enable-flow-logs
+            resource: vpc
+            filters:
+              - type: flow-logs
+                enabled: false
+            actions:
+              - type: set-vpc-flow
+                DeliverLogsPermissionArn: arn:iam:role
+                LogGroupName: /custodian/vpc/flowlogs/
+                TrafficType: ALL
+    """
+    permissions = ('ec2:CreateFlowLogs',)
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['set-vpc-flow']},
+            'DeliverLogsPermissionArn': {'type': 'string'},
+            'LogGroupName': {'type': 'string'},
+            'TrafficType': {'type': 'string',
+                            'enum': ['ACCEPT', 'REJECT', 'ALL']},
+            'required': ['DeliverLogsPermissionArn', 'LogGroupName',
+                         'ResourceIds']
+        }
+    }
+
+    def validate(self):
```

Comment:
this validate method appears to be superflous given the jsonschema for the action.